src/prpy/base/barretthand.py

```diff
@@ -62,24 +66,28 @@ def __init__(self, sim, manipulator, owd_namespace, bhd_namespace, ft_sim=True):
 
         manipulator.SetChuckingDirection(closing_direction)
 
-        # Load the  hand controller.
         robot = self.manipulator.GetRobot()
         env = robot.GetEnv()
-        self.controller = robot.AttachController(name=self.GetName(),
-            args='BHController {0:s} {1:s}'.format('prpy', bhd_namespace),
-            dof_indices=self.GetIndices(), affine_dofs=0, simulated=sim)
 
-        # Hand state, force/torque sensor, and tactile pads.
+        # Controller setup
+        self.bhd_namespace = bhd_namespace
+        self.hand_side = bhd_namespace[1:]  # TODO hand name hack
         if not sim:
-            self.handstate_sensor = util.create_sensor(env,
-                'HandstateSensor {0:s} {1:s}'.format('prpy', bhd_namespace))
+            self.controller = PositionCommandController('', self.hand_side +
+                                                        '_hand_controller')
+        else:
+            self.controller = robot.AttachController(name=self.GetName(),
+                                                     args='IdealController',
+                                                     dof_indices=self.GetIndices(),
+                                                     affine_dofs=0,
+                                                     simulated=sim)
 
         self.ft_simulated = ft_sim
-        if not ft_sim:
-            self.ft_sensor = util.create_sensor(env,
-                'BarrettFTSensor {0:s} {1:s}'.format('prpy', owd_namespace))
+        self.ft_tare_controller = TriggerController('', self.hand_side +
+                                                    '_tare_controller',
+                                                    ft_sim)
```

Comment:
TriggerController handles the sim case internally, since we have no other "ideal controller" for the tare controller.